### PR TITLE
Update readme for create and deploy roles

### LINKED ISSUE #135: Description of Create vs. Deploy role incorrect
Referring to "Understanding our Ansible roles" in the collection README/documentation, I think the description of the Create and Deploy role is not entirely correct. The documentation says "The deploy role is going to deploy those changes to the NDFC controller. [...] This is the role that is going to make the changes in the NDFC controller." 

However, I observe the Create role already applies the changes to the NDFC controller. The Deploy role applies the changes to the switches.

https://github.com/netascode/ansible-dc-vxlan
https://galaxy.ansible.com/ui/repo/published/cisco/nac_dc_vxlan/docs/

### PR DESCRIPTION
**Pull request by mtarking**
Update readme for create and deploy roles to resolve #135

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,21 +20,21 @@ This is achieved by creating YAML files that contain a pre-determined data schem
...
 
 ### Create role
 
 Role: [cisco.nac_dc_vxlan.dtc.create](https://github.com/netascode/ansible-dc-vxlan/blob/develop/roles/dtc/create/README.md)
 
-This role is going to create all the templates and variable parameters that are going to be used in the deployment of the VXLAN fabric. This role converts the data model into the proper templates that are required by the Ansible module to be able to communicate with the NDFC controller.
+This role is going to use a templates and variable parameters from the data model to create and stage all the configuration for your VXLAN fabric. This role uses the data model in the proper templates to render the expected data that is required by the NDFC Ansible modules.
```
Comment (marehler) [inline README.md:31]: There are two steps. First, this role creates the configuration from the templates and variables. (Apparently by including nac_dc_vxlan/roles/dtc/common/tasks/main.yml, although I don't get how it is included.) Second, this roles applies the configuration to the NDFC, thereby creating the fabric on the NDFC. (By including nac_dc_vxlan/roles/dtc/create/tasks/main.yml.)

File: README.md [hunk 2]
```diff
@@ -20,21 +20,21 @@ This is achieved by creating YAML files that contain a pre-determined data schem
...
 
 ### Deploy role
 
 Role: [cisco.nac_dc_vxlan.dtc.deploy](https://github.com/netascode/ansible-dc-vxlan/blob/develop/roles/dtc/deploy/README.md)
 
-The deploy role is going to deploy those changes to the NDFC controller. This role is going to take the templates and variable parameters that were created in the `create` role and deploy them to the NDFC controller. This is the role that is going to make the changes in the NDFC controller.
+The deploy role is going to recalculate and deploy the staged changes from the `create` role to your NDFC controller.
```
Comment (marehler) [inline README.md:37]: No, the deploy roles triggers NDFC to recalculate and deploy the configuration (which already has been applied to NDFC by the create role) to the switches.

**Review by marehler — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (mtarking) [reply README.md:31]: Hi @marehler, correct, but I don't think we need explicitly expand on all the dependencies and details under the covers as that's the point in the collection in taking that burden off the consumer. Essentially, one should not need to dive into the role. 

Also agree with your second sentence and that is what is meant by staging the configuration in NDFC. Create stages and does not push to the switches (except in a few one off scenarios to ensure certain things are in a working order for the collection to run). 

Let me know if that doesn't make sense or if we're crossing wires.

File: README.md (hunk 2 above)
Comment (mtarking) [reply README.md:37]: Again, staged in fact means staging the configuration within NDFC, so I believe we're saying the same thing, no?

File: README.md (hunk 1 above)
Comment (marehler) [reply README.md:31]: Maybe the term "stage" is a bit unclear. Besides it says "stage", not "stage in NDFC". Therefore I thought it meant just creating and preparing the configuration in Ansible but not pushing them yet to NDFC. How about rephrasing the first sentence to "[...] to create the configuration in NFDC"? Then it should be clear.

File: README.md (hunk 2 above)
Comment (marehler) [reply README.md:37]: Currently it says "deploy to NDFC", but it's actually "deploy from NDFC to the switches". Therefore I thought it meant creating the configuration in NDFC what is done in the create role. I'd suggest to rephrase it like "The deploy role triggers NDFC to recalculate and deploy the configuration to the switches."

File: README.md [hunk 3]
```diff
@@ -20,21 +20,21 @@ This is achieved by creating YAML files that contain a pre-determined data schem
 
 Role: [cisco.nac_dc_vxlan.validate](https://github.com/netascode/ansible-dc-vxlan/blob/develop/roles/validate/README.md)
 
-The validate role function is to ensure that the data model is correct, and that the data model is going to be able to be processed by the subsequent roles. The validate role is going to read all the files in the `host_vars` directory and create a single data model in memory for execution.
+The validate role has the function to ensure that the data model is correct, and that the data model is going to be able to be processed by the subsequent roles. The validate role is going to read all the files in the `host_vars` directory that is named after the inventory group defined in your inventory file and create a single data model in memory for execution.
```
Comment (mikewiebe) [inline README.md:23]: The documentation in our example repo is more accurate so let's use that language here.

File: README.md (hunk 1 above)
Comment (mikewiebe) [reply README.md:31]: The documentation in our example repo is clearer so I propose we use that here as well.

File: README.md (hunk 2 above)
Comment (mikewiebe) [reply README.md:37]: The documentation in our example repo is clearer so I propose we use that here as well.

File: README.md (hunk 1 above)
Comment (marehler) [reply README.md:31]: Agree, that is clearer.

**Comment by mtarking** [comment]
> Looks good now.

Thanks @marehler!

**Pull request merged**